### DISCUSSION TITLE
present next trial based on condition

### DISCUSSION THREAD

**Discussion by Stajduhar-A** [jspsych/jsPsych#776] [Q&A] [answered]
Hi Josh,

I would like to dynamically determine which trial will occur next based on the fulfillment of some condition. For instance, I want a survey trial to be presented next to the participant if the participant pressed the letter 'f'. I want the experiment to bypass the survey trial and instead progress to the next image trial if the participant pressed 'k'. Is there any way to pull out the key_press data from a previous trial (i.e. timeline object), evaluate the selection made previously in this timeline object, and then have the new timeline object execute either option A (presenting the survey trial) or option B (moving on to the next image trial as planned)? I have tried the if(data.key_press == 75){//do something} function I saw on the jsPsych website, however, it doesn't appear to work if placed within a new timeline object. Any help would be appreciated! 

Thanks,
A

**Comment by jodeleeuw** [comment, marked as answer]
Sounds like the survey trial should be wrapped in [a conditional timeline](https://www.jspsych.org/overview/timeline/#conditional-timelines). The example on the docs page is very similar to what you need to do. Copying it here for ease of reference:

```js
var pre_if_trial = {
    type: 'html-keyboard-response',
    stimulus: 'The next trial is in a conditional statement. Press S to skip it, or V to view it.'
}

var if_trial = {
    type: 'html-keyboard-response',
    stimulus: 'You chose to view the trial. Press any key to continue.'
}

var if_node = {
    timeline: [if_trial],
    conditional_function: function(){
        // get the data from the previous trial,
        // and check which key was pressed
        var data = jsPsych.data.get().last(1).values()[0];
        if(data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode('s')){
            return false;
        } else {
            return true;
        }
    }
}

var after_if_trial = {
    type: 'html-keyboard-response',
    stimulus: 'This is the trial after the conditional.'
}

jsPsych.init({
    timeline: [pre_if_trial, if_node, after_if_trial],
    on_finish: function(){jsPsych.data.displayData(); }
});
```

**Reply by Stajduhar-A** [reply]
Thanks so much, Josh! That was exactly the solution I was looking for. 

I have another question if you don't mind. I want to randomize my trials now and I saw you can randomize objects using the jsPsych.randomization() function... however, I have the type parameter in each trial. I noticed that the objects you have as examples don't have this parameter... is there any way to get rid of it in the trial objects, randomize, and still have it work without type being specified? I'm curious how to go about that. Thanks again!

**Reply by jodeleeuw** [reply]
Would you mind posting this as a new discussion? It'll make it easier for others to find in the future if folks are having a similar question. Thanks!

**Reply by Stajduhar-A** [reply]
Absolutely!
